rsconf/component/sirepo_sandbox.py

```diff
@@ -0,0 +1,125 @@
+# -*- coding: utf-8 -*-
+"""Sirepo development in a docker container on a remote machine
+
+:copyright: Copyright (c) 2019 RadiaSoft LLC.  All Rights Reserved.
+:license: http://www.apache.org/licenses/LICENSE-2.0.html
+"""
+from pykern import pkio
+from pykern.pkcollections import PKDict
+from pykern.pkdebug import pkdp
+from rsconf import component
+import subprocess
+
+
+class T(component.T):
+    def install_resource(self, *args, **kwargs):
+        super().install_resource(*args, component_name="sirepo_sandbox", **kwargs)
+
+    def internal_build_compile(self):
+        from rsconf import systemd
+
+        if self.name == "sirepo_sandbox":
+            for u in self.hdb.sirepo_sandbox.users.keys():
+                t = T(u, self.buildt)
+                self.buildt.build_component(t)
+            return
+        self.buildt.require_component("docker", "network")
+        jc, _ = self.j2_ctx_init()
+        z = jc.sirepo_sandbox
+        z.setdefault("volumes", {})
+        # TODO(e-carlin): Better way then building the path manually
+        # /srv/<component-name> is output by systemd.docker_unit_prepare
+        # below but to call that we need z.host.ssh_d. z.host needs z.host_d
+        # so we are in a dependency cycle
+        z.host_d = z.host_root_d.join(self.name, "db")
+        z.secrets = self._gen_keys(jc, z)
+        for x in "guest", "host":
+            z[x] = self._gen_paths(z, z.get(x + "_d"))
+        z.run_u = jc.rsconf_db.run_u
+        # Only additional config for the server is the sshd config.
+        z.run_d = systemd.docker_unit_prepare(self, jc, watch_files=[z.host.ssh_d])
+        self._network(jc, z)
+
+    def internal_build_write(self):
+        from rsconf import systemd
+
+        if self.name == "sirepo_sandbox":
+            self.append_root_bash(": nothing to do")
```

Comment:
rather ": user instances do all the installs"

Reply:
af3b1f8